src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -38,24 +40,30 @@ def __init__(self, model, config_list):
         """
         self.bound_model = model
         self.config_list = config_list
-        self.modules_to_compress = None
-        self.modules_wrapper = None
-        self.buffers = {}
+        self.optimizer = optimizer
+
+        self.modules_to_compress = self.detect_modules_to_compress()
+        self.modules_wrapper = []
         self.is_wrapped = False
 
+        for layer, config in self.modules_to_compress:
+            wrapper = self._wrap_modules(layer, config)
+            self.modules_wrapper.append(wrapper)
+
+        self._wrap_model()
+
     def detect_modules_to_compress(self):
         """
         detect all modules should be compressed, and save the result in `self.modules_to_compress`.
         The model will be instrumented and user should never edit it after calling this method.
         """
-        if self.modules_to_compress is None:
-            self.modules_to_compress = []
-            for name, module in self.bound_model.named_modules():
-                layer = LayerInfo(name, module)
-                config = self.select_config(layer)
-                if config is not None:
-                    self.modules_to_compress.append((layer, config))
-        return self.modules_to_compress
+        modules_to_compress = []
```

Comment:
where will this function be called? if this function is called only in __init__, please make it private

Reply:
and why remove the check `if self.modules_to_compress is None:` ?

Reply:
1 only in `__init__`  
2 because self.modules_to_compress is initialized in ` __init__`, which means it must be a list

Reply:
1. please add underline at the beginning of this function name
2. what if this function is called at other places such as pruning algorithm? if this function is only allowed to be called once, suggest to add that check, in order to avoid some unexpected behavior when this function is accidentally called multiple times

Reply:
good suggestion!